R/utils.R

```diff
@@ -53,12 +53,33 @@ touchstone_clear <- function(all = FALSE) {
   fs::dir_delete(paths)
 }
 
-#' Evaluate an expression
+#' Evaluate an expression for sideeffects
 #'
-#' @param text Character vector with code to evaluate.
+#'
+#' @param ... Character vector  of length 1 or expression with code to evaluate. This will be quoted using
+#' [rlang::enexprs()], so you can use `!!`.
+#' @param env Environment in which the expression will be evaluated.
+#' @return The quoted input (invisibly).
 #' @keywords internal
-exprs_eval <- function(...) {
-  eval(parse(text = unlist(rlang::list2(...))))
+exprs_eval <- function(..., env = parent.frame()) {
+    
+    expr <- rlang::enexprs(...)[[1]]
```

Comment:
maybe we should use `enquos()` instead? Not so sure about the differences, but apparently `enquos()` evaluates in the environment in which the expression was created. Also see `rlang::eval_tidy()`, argument `env`, where you'd not have to supply an environment I think because the input would be a quosure.

Reply:
I was not sure about this either. I think if we use quosures it could lead to issues with passing code as strings:
![image](https://user-images.githubusercontent.com/16141871/130791054-740e584b-b83a-4c05-bc90-011776f01efe.png)
Or at least lead to more work to check if it is a string and then supply the default environment again. I am not sure what the benefit over expressions would be?

Reply:
We could restructure to use quosures and drop the `enexpr` within. That would allow users to access variables defined in `script.R` that are not explicitly passed into `benchmark_run_ref` as far as I understand. That would mean more flexibility but also make it less clear what is involved in which test.

Reply:
Ok, I read up on it a bit, quosures wont work because assignments don't work: